core/dynamo.c

```diff
@@ -978,7 +984,22 @@ dynamo_shared_exit(thread_record_t *toexit /* must ==cur thread for Linux */
      * client trying to use api routines that depend on fragment state.
      */
     instrument_exit();
-#endif
+# ifdef CLIENT_SIDELINE
+    /* We only need do a second synch-all if there are sideline client threads. */
```

Comment:
The comment implies there should be a conditional here: can we avoid the cost of the locks, etc. in the common case, which is zero sideline threads, by having a get_num_threads() > 1 check here?